AEXS/aex-141.md

```diff
@@ -490,11 +492,254 @@ stateful entrypoint burn(token_id: int) : unit
+    entrypoint template_supply : () => int
+
+    stateful entrypoint create_template : (metadata, option(int)) => int
+    stateful entrypoint delete_template : (int) => unit
+    stateful entrypoint template_mint : (address, int, option(string)) => int
+    stateful entrypoint decrease_edition_limit : (int, int) => unit
+```
+
+### template_metadata_type\(\)
+
+Returns the metadata type which is used for templates.
+
+```sophia
+entrypoint template_metadata_type() : template_metadata_type
+```
+
+| return | type |
+| :--- | :--- |
+| template_metadata_type | template_metadata_type |
+
+### template\(\)
+
+Returns the template for the provided `template_id` in case it exists, otherwise `None`.
+
+```sophia
+entrypoint template(template_id: int) : option(template)
+```
+
+| return | type |
+| :--- | :--- |
+| template | option(template) |
+
+### template_supply\(\)
+
+Returns the total amount of templates that currently exist.
+
+```sophia
+entrypoint template_supply() : ínt
+```
+
+| return | type |
+| :--- | :--- |
+| template_supply | int |
+
+### create_template\(\)
+
+Creates a new template for specific immutable metadata. The `immutable_metadata` needs to be `MetadataIdentifier` in case `template_metadata_type` is `T_URL` and `MetadataMap` for `T_MAP`. The edition limit defines how many NFTs can be minted based on a specific template. If no edition limit is provided, an unlimited amount of NFTs can be created using the template. The edition limit can be decreased at any time.
+
+Emits the `TemplateCreation` event.
+
+Note: It's recommended to perform a check if `immutable_metadata` is considered valid according to individual requirements.
+
+Throws in case the wrong variant for `metadata` is provided:
+- variant MUST be `MetadataIdentifier` if `template_metadata_type` is `T_URL`
+- variant MUST be `MetadataMap` if `template_metadata_type` is `T_MAP`
+
+```sophia
+stateful entrypoint create_template(immutable_metadata: metadata, edition_limit: option(int)) : int
+```
+
+| parameter | type |
+| :--- | :--- |
+| immutable_metadata | metadata |
+| edition_limit | option(int) |
+
+| return | type |
+| :--- | :--- |
+| template_id | int |
+
+### delete_template\(\)
+
+Deletes the template with the provided id.
+
+Emits the `TemplateDeletion` event.
+
+Throws if:
+
+- the provided `template_id` does not exist
+- an NFT based on this template has already been created
+
+```sophia
+stateful entrypoint delete_template(template_id: int) : unit
+```
+
+| parameter | type |
+| :--- | :--- |
+| template_id | int |
+
+### template_mint\(\)
+
+Mints a new NFT for the provided template id.
+
+Emits the `TemplateMint` event.
+
+Throws if:
+
+- the provided `template_id` does not exist
+- the mint would exceed the current `edition_limit` of the template
+- the call to `IAEX141Receiver.on_aex141_received` implementation failed (safe transfer)
+
+```sophia
+stateful entrypoint template_mint(to: address, template_id: int, data: option(string)) : int
```

Comment:
just stumbled over this `data` that we also have in the base standard, maybe this should be better explained

Reply:
added info about `data`: https://github.com/aeternity/AEXs/pull/149/commits/bbc5df5a1f0cc01e2631ee75dbd8104972ae10af